### PR TITLE
[Validator] Pre-check constraint validator dependencies

### PR DESCRIPTION
**Pull request by ro0NL**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no     
| Deprecations? | yes
| Tests pass?   | yes    
| Fixed tickets | #25865
| License       | MIT
| Doc PR        | symfony/symfony-docs#...

### CODE REVIEW

File: UPGRADE-4.2.md [hunk 1]
```diff
@@ -216,4 +216,6 @@ Validator
  * The component is now decoupled from `symfony/translation` and uses `Symfony\Contracts\Translation\TranslatorInterface` instead
  * The `ValidatorBuilderInterface` has been deprecated and `ValidatorBuilder` made final
  * Deprecated validating instances of `\DateTimeInterface` in `DateTimeValidator`, `DateValidator` and `TimeValidator`. Use `Type` instead or remove the constraint if the underlying model is type hinted to `\DateTimeInterface` already.
- * Using the `Bic` constraint without `symfony/intl` is deprecated
+ * Using the `Bic`, `Country`, `Currency`, `Language` and `Locale` constraints without `symfony/intl` is deprecated
```
Comment (xabbuh) [inline UPGRADE-4.2.md]: missing final dots here and below (same below for other upgrade files)

Comment (ro0NL) [reply UPGRADE-4.2.md]: i think we only do that for multiple sequences...

File: src/Symfony/Component/Validator/Constraints/BicValidator.php [hunk 2]
```diff
@@ -68,11 +69,12 @@ public function validate($value, Constraint $constraint)
             return;
         }
 
-        // @deprecated since Symfony 4.2
+        // @deprecated since Symfony 4.2, will throw in 5.0
         if (class_exists(Intl::class)) {
             $validCountryCode = isset(Intl::getRegionBundle()->getCountryNames()[substr($canonicalize, 4, 2)]);
         } else {
             $validCountryCode = ctype_alpha(substr($canonicalize, 4, 2));
+            // throw new LogicException('The "symfony/intl" component is required to use the Bic constraint.');
```
Comment (xabbuh) [inline src/Symfony/Component/Validator/Constraints/BicValidator.php]: We should trigger a deprecation here for now (at least when the passed constraint is a subclass of the `Bic` constraint).

Comment (chalasr) [reply src/Symfony/Component/Validator/Constraints/BicValidator.php]: agreed, sounds like a very edge case afterall

Comment (xabbuh) [reply src/Symfony/Component/Validator/Constraints/BicValidator.php]: fair enough, but I think we should still throw the exception in 5.0

**Review by xabbuh — APPROVED** [review]

**Review by chalasr — APPROVED** [review]

**Review by nicolas-grekas — APPROVED** [review]

**Comment by nicolas-grekas** [comment]
Thank you @ro0NL.

**Pull request merged**